lib/crypto/crypto_compat.c

```diff
@@ -23,7 +23,7 @@ crypto_compat_RSA_valid_size(const RSA * const rsa)
 	/* Sanity checks. */
 	assert(rsa != NULL);
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
```

Comment:
How about putting

```
/* LibreSSL claims to be OpenSSL 2.0, but (currently) has APIs compatible with OpenSSL 1.0.0. */
#ifdef LIBRESSL_VERSION_NUMBER
#undef OPENSSL_VERSION_NUMBER
#define OPENSSL_VERSION_NUMBER 0x100000000L
#endif
```

at the top of the file?  Simpler than checking for LibreSSL in four different places...